include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -282,10 +286,42 @@ struct alignment_configurator
         if constexpr (config_t::template exists<align_cfg::band>())
             throw invalid_alignment_configuration{"Banded alignments are yet not supported."};
 
-        if constexpr (config_t::template exists<align_cfg::aligned_ends>())
-            throw invalid_alignment_configuration{"Free end gaps is not yet supported."};
+        // ----------------------------------------------------------------------------
+        // Configure semi-global alignment
+        // ----------------------------------------------------------------------------
+
+        // Get the value for the sequence ends configuration.
+        auto align_ends_cfg = cfg.template value_or<align_cfg::aligned_ends>(align_cfg::none_ends_free);
+        using align_ends_cfg_t = remove_cvref_t<decltype(align_ends_cfg)>;
+
+        auto configure_semi_global = [&](auto is_semi_global)
```

Comment:
I would name this differently

Maybe

```suggestion
        auto configure_edit_traits = [&](auto is_semi_global)
```